src/analysis/processing/qgsalgorithmremoveduplicatesbyattribute.cpp

```diff
@@ -0,0 +1,171 @@
+/***************************************************************************
+                         qgsalgorithmremoveduplicatesbyattribute.cpp
+                         ----------------------------------
+    begin                : October 2018
+    copyright            : (C) 2018 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmremoveduplicatesbyattribute.h"
+
+///@cond PRIVATE
+
+QString QgsRemoveDuplicatesByAttributeAlgorithm::name() const
+{
+  return QStringLiteral( "removeduplicatesbyattribute" );
+}
+
+QString QgsRemoveDuplicatesByAttributeAlgorithm::displayName() const
+{
+  return QObject::tr( "Remove duplicates by attribute" );
+}
+
+QStringList QgsRemoveDuplicatesByAttributeAlgorithm::tags() const
+{
+  return QObject::tr( "field,value,same,filter" ).split( ',' );
+}
+
+QString QgsRemoveDuplicatesByAttributeAlgorithm::group() const
+{
+  return QObject::tr( "Vector selection" );
```

Comment:
A very similar algorithm, delete duplicate geometries, is found in the Vector general group, maybe this new algorithm should also be placed in that group. IMHO it'd be a better fit.